src/app/content/practiceQuestions/reducer.ts

```diff
@@ -21,23 +21,18 @@ export const initialState: State = {
 const reducer: Reducer<State, AnyAction> = (state = initialState, action): State => {
   switch (action.type) {
     case getType(locationChange):
-      const shouldBeOpen = action.payload.query[modalQueryParameterName] === modalUrlName
-        && action.payload.action === 'PUSH';
+      const shouldBeOpen = action.payload.query[modalQueryParameterName] === modalUrlName;
```

Comment:
I think this will not be that simple, because not every book has Practice questions so if someone enters the URL with `modal=PQ` for a book that does not have PQ then we shouldn't show the modal.

Reply:
exactly, this is a problem to solve
Maybe we should remove `open` property from the redux state and instead have `isOpen` selector that will look for `modal=PQ` and `bookHasPracticeQuestions` etc?

Reply:
But does it fix the timing issue? `bookHasPracticeQuestions` will be `false` during the `locationChange` action.

Reply:
selector should change when it will become true

Reply:
I've meant remove `open` from this reducer and change `practiceQuestionsOpen` selector to look for `modal: PQ` in `navigation.query` + `hasPracticeQuestions` selector
You should be able to do something like:
```ts
export const practiceQuestionsOpen = createSelector(
  hasPracticeQuestions,
  navigationState
  (hasPQ) => hasPQ && navigationState.query && navigationState.query.modal === 'PQ'
);
```